src/canmatrix/canmatrix.py

```diff
@@ -1278,9 +1506,10 @@ def merge(self, mergeArray):
                         "Name Conflict, could not copy/merge EnvVar " + envVar)
 
     def setFdType(self):
```

Comment:
This sounds more like `guessIfFdAndSet()` or somesuch.  @ebroecker, what's the history on this?  We can't tell if a frame is FD or not?  Or can you not have an FD frame with <= 8 bytes?  Why only setting to `True` and no case for setting `False`?

Reply:
you are right - this is only guessing.
Meanwhile with J1939 this may even do mistakes.
In deed I think FD has long DLCs only - but I may be wrong. If I am wrong, setting False would be wrong also :-)

Reply:
I thing the data speed depends merely on the flag in the CAN Frame header on the bus. That means "DLC>8 implies CAN FD" but DLC <= 8 implies nothing.

The function could than rather be `checkFdRequired()`. And set back to False, if there is nothing with long DLC. Agree?

Reply:
Agree

Reply:
I'll propose it separately so that this "big" PR contains minimum changes.

Reply:
As is this could probably be at least 3 I think.  :]  Every new change that's added holds up all the rest from being accepted.